craft_providers/bases/buildd.py

```diff
@@ -421,6 +424,33 @@ def _disable_automatic_apt(
             file_mode="0644",
         )
 
+    def _disable_and_wait_for_snap_refresh(
+        self, *, executor: Executor, deadline: Optional[float]
+    ) -> None:
+        """Disable automatic snap refreshes and wait for refreshes to complete.
+
+        Craft-providers manages the installation and versions of snaps inside the
+        build environment, so automatic refreshes of snaps by snapd are disabled.
+        """
+        # disable refresh for 1 day
```

Comment:
Is 1 day long enough? (otoh if core 18-20 are using this it's probably better to stay consistent)

Reply:
If an craft lifecycle runs for more than 24 hours, then perhaps not.

I do agree that the consistency may be better.  There is no comment in core18|20 code for why 24 hours was chosen.